src/tarantool_schema.c

```diff
@@ -557,6 +557,24 @@ tarantool_schema_get_sid_by_string(
 	return space->space_number;
 }
 
+int32_t
+tarantool_schema_get_sid_by_number(
+		struct tarantool_schema *schema_obj,
+		uint32_t sid
+) {
+	struct mh_schema_space_t *schema = schema_obj->space_hash;
+	struct schema_key space_key = {
+		(void *)&sid,
+		sizeof(uint32_t), 0
+	};
```

Comment:
Cast of `uint32 *` to `const char *` (via `void *`). Looks very strange.

Reply:
Got it. Both numeric and string values are compared just as byte arrays without taking its types into account.

A clash is possible in theory: say, valid space name `!!!!` is the same as a [valid](https://www.tarantool.io/en/doc/1.10/book/box/limitations/) space id 555819297 (0x21212121). The minimal uint32_t number that corresponds to a valid space name (four spaces) is 538976288 (0x20202020) if we'll choose printable characters from ASCII. I'm not sure whether it is possible to find a clash with lower space id using unicode printable symbols.

It is not sane, but it seems it should work in practice.

BTW, just observed that the last structure field (`number`) is not left uninitialized. It is initialized as zero, but the field is not used anywhere, so it does not matter. If we'll use it, however, it seems logical to set it in the similar way as other code do (see `schema_add_index()`). I mean, set it to a space id.

Reply:
Nothing to do, closing.